metrics/frugalscore/frugalscore.py

```diff
@@ -0,0 +1,96 @@
+  year={2021}
+}
+"""
+
+_DESCRIPTION = """\
+FrugalScore is a reference-based metric for NLG models evaluation. It is based on a distillation approach that allows to learn a fixed, low cost version of any expensive NLG metric, while retaining most of its original performance.
+"""
+
+
+_KWARGS_DESCRIPTION = """
+Calculates how good are predictions given some references, using certain scores
+Args:
+    predictions (list of str): list of predictions to score. Each predictions
+        should be a string.
+    references (list of str): list of reference for each prediction. Each
+        reference should be a string.
+    pretrained_model_name_or_path (str): the pretrained model used to generate the scores.
+    batch_size (int): the batch size for predictions.
+    max_length (int): maximum sequence length.
+Returns:
+    scores (list of int): list of scores.
+Examples:
+    >>> my_new_metric = datasets.load_metric("frugalscore")
+    >>> results = my_new_metric.compute(predictions=['hello there', 'huggingface'], references=['hello world', 'hugging face'])
+    >>> print(results)
+    {'scores': [0.631, 0.645]}
+"""
+
+
+@datasets.utils.file_utils.add_start_docstrings(_DESCRIPTION, _KWARGS_DESCRIPTION)
+class FRUGALSCORE(datasets.Metric):
+    def _info(self):
+        return datasets.MetricInfo(
+            description=_DESCRIPTION,
+            citation=_CITATION,
+            inputs_description=_KWARGS_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "predictions": datasets.Value("string"),
+                    "references": datasets.Value("string"),
+                }
+            ),
+            homepage="https://github.com/moussaKam/FrugalScore",
+        )
+
+    def _compute(
+        self,
+        predictions,
+        references,
+        pretrained_model_name_or_path="moussaKam/frugalscore_tiny_bert-base_bert-score",
+        batch_size=32,
+        max_length=128,
+    ):
+        """Returns the scores"""
+        model = AutoModelForSequenceClassification.from_pretrained(pretrained_model_name_or_path)
+        tokenizer = AutoTokenizer.from_pretrained(pretrained_model_name_or_path)
+
+        training_args = TrainingArguments("trainer", fp16=True, per_device_eval_batch_size=batch_size, report_to=None)
+        dataset = {"sentence1": predictions, "sentence2": references}
+        raw_datasets = datasets.Dataset.from_dict(dataset)
+
+        def tokenize_function(data):
+            return tokenizer(
+                data["sentence1"], data["sentence2"], max_length=max_length, truncation=True, padding=True
+            )
+
+        tokenized_datasets = raw_datasets.map(tokenize_function, batched=True)
+        tokenized_datasets.remove_columns(["sentence1", "sentence2"])
+        trainer = Trainer(model, training_args, tokenizer=tokenizer)
+        predictions = trainer.predict(tokenized_datasets)
+        return {"scores": list(predictions.predictions.squeeze(-1))}
```

Comment:
Here would it make sense to also return the mean of the scores ?

Reply:
I think it's OK like that, I tried to follow [`bleurt`](https://github.com/huggingface/datasets/blob/master/metrics/bleurt/bleurt.py#L124)

Reply:
Ok ! Sounds good :)